kong/runloop/plugins_iterator.lua

```diff
@@ -169,55 +169,42 @@ end
 -- @tparam string|nil consumer_id The consumer identifier.
 -- @return any|nil The configuration corresponding to the best matching combination, or 'nil' if no configuration is found.
 ---
-function PluginsIterator.lookup_cfg(combos, route_id, service_id, consumer_id)
-  -- Use the build_compound_key function to create an index for the 'combos' table
-  local build_compound_key = PluginsIterator.build_compound_key
-
-    local key
-    if route_id and service_id and consumer_id then
-        key = build_compound_key(route_id, service_id, consumer_id)
-        if combos[key] then
-            return combos[key]
-        end
-    end
-    if route_id and consumer_id then
-        key = build_compound_key(route_id, nil, consumer_id)
-        if combos[key] then
-            return combos[key]
-        end
-    end
-    if service_id and consumer_id then
-        key = build_compound_key(nil, service_id, consumer_id)
-        if combos[key] then
-            return combos[key]
-        end
-    end
-    if route_id and service_id then
-        key = build_compound_key(route_id, service_id, nil)
-        if combos[key] then
-            return combos[key]
-        end
-    end
-    if consumer_id then
-        key = build_compound_key(nil, nil, consumer_id)
-        if combos[key] then
-            return combos[key]
-        end
-    end
-    if route_id then
-        key = build_compound_key(route_id, nil, nil)
-        if combos[key] then
-            return combos[key]
-        end
+local function lookup_cfg(combos, route_id, service_id, consumer_id)
+  local ids = { route_id, service_id, consumer_id, }
+
+  local orders = {
+    { true, true, true, },  -- Route, Service, Consumer
+    { true, nil , true, },  -- Route,        , Consumer
+    { nil , true, true, },  --      , Service, Consumer
+    { true, true, nil , },  -- Route, Service,
+    { nil , nil , true, },  --      ,        , Consumer
+    { true, nil , nil , },  -- Route,        ,
+    { nil , true, nil , },  --      , Service,
+  }
```

Comment:
I am not sure about this PR, but can we reduce the table creation by moving this to module level?

Reply:
good idea, thanks.